merlin/models/tf/utils/tf_utils.py

```diff
@@ -469,3 +469,29 @@ def list_col_to_ragged(col: Tuple[tf.Tensor, tf.Tensor]):
         row_lengths = tf.cast(row_lengths, tf.int32)
 
     return tf.RaggedTensor.from_row_lengths(values, row_lengths)
+
+
+def check_inputs_mask_compatible_shape(
+    inputs: Union[tf.Tensor, tf.RaggedTensor], mask: Union[tf.Tensor, tf.RaggedTensor]
+):
+    """Checks if the shape and the type of the input and mask tensors are compatible.
+    Parameters
+    ----------
+    inputs : Union[tf.Tensor, tf.RaggedTensor]
+        The input tensor, which can be either a dense or ragged tensor.
+    mask : Union[tf.Tensor, tf.RaggedTensor]
+        The mask tensor, which can be either a dense or ragged tensor.
+
+    Returns
+    -------
+    bool: True if the shape of the input and mask tensors are compatible, False otherwise.
+    """
+    result = False
+    if type(inputs) == type(mask) and (inputs.shape.as_list()[:-1] == mask.shape.as_list()):
```

Comment:
Thanks for your comment! The function was originally implemented in `ReplaceMaskedembeddings` to check the 2D mask because it currently only supports 3-D sequential inputs. I agree it would be more useful to make the function more generic to support inputs of different dimensions. 
I updated the code to use `inputs.shape.as_list()[:len(mask.shape)]` instead of inputs.shape.as_list()[:-1] to make the function more flexible.